test-basic/data-movement-request.js

```diff
@@ -50,15 +45,25 @@ function createDocumentDescriptors() {
         readable.push(temp);
     }
     readable.push(null);
+
 }
 
-function writeDocs(record, done) {
-    if(record === null) {
-        return done();
+function writeDocs(writerId, done) {
+
+    const writeBatchArray = [];
+    for(let i=0; i<batchSize; i++) {
+        const record = readable.read();
+        if(record === null) {
+            if(writeBatchArray.length!==0) {
+                dbWriter.documents.write(writeBatchArray);
+            }
```

Comment:
If the last batch is partial, will writeDocs() return before it is written?

Instead, would it work to break out of the loop instead of returning and have a branch after the loop similar to the following:

    switch(writeBatchArray.length) {
    case 0: ...call done() immediately and break...
    case batchSize:  ...call documents.write() with the writeDocs() callback and break...
    default: ...call documents.write() with done() as the callback and break...
    }

Reply:
Done